ui/k-toolbar/main.kytos

```diff
@@ -57,35 +63,125 @@ module.exports = {
+//                "component": 'list_connections',
+                "content": {},
+                "icon": "desktop",
+                "title": "View Connections",
+                "subtitle": "by kytos/mef_eline"
+            }
+            this.$kytos.$emit("showInfoPanel", content)
+        },
+
     request_circuit () {
      var request = {
         "name" : this.circuit_name,
         "uni_a": {"interface_id": this.endpoint_a},
         "uni_z": {"interface_id": this.endpoint_z}
      }
-
      if (this.tag_type_a != "" && this.tag_value_a != ""){
-        request.uni_a['tag'] = {tag_type: this.tag_type_a,
-                                tag_value: this.tag_value_a}
+        request.uni_a['tag'] = {tag_type: parseInt(this.tag_type_a),
+                                value: parseInt(this.tag_value_a)}
      }
      if (this.tag_type_z != "" && this.tag_value_z != ""){
-        request.uni_z['tag'] = {tag_type: this.tag_type_z,
-                                tag_value: this.tag_value_z}
+        request.uni_z['tag'] = {tag_type: parseInt(this.tag_type_z),
+                                value: parseInt(this.tag_value_z)}
      }
-     $.ajax({
-       url: this.$kytos_server_api + "kytos/mef_eline/v2/evc/",
-       type:"POST",
-       data: JSON.stringify(request),
-       dataType: "json",
-       contentType: "application/json; charset=utf-8",
-       success: function(data, textStatus){
-           console.log('Circuit with id '+data.circuit_id+" was creaded.")
-       },
-       error: function(data){
-           console.log("Error ("+data.status+"): "+ data.responseJSON)
-       },
+     //variable to check if there is no endpoint in use
+     var endpointsInUSe = false;
+     var connectionCreated = false;
+
+     //used to notify the user of changes that were made
+
+     let notification = {
+        icon:'gear'
+     };
+
+     $.ajax(
+     //GET Request to check availability
+     {
+        url: this.$kytos_server_api + "kytos/mef_eline/v2/evc/",
+        type:"GET",
+        data: JSON.stringify(),
+        dataType: "json",
+        contentType: "application/json; charset=utf-8",
+        success: function(data, textStatus){
+
+            $.each(data, function(i , evc){
+                //is request endpoint A port in used
+                var request_port_a_InUse1 = (request.uni_a.interface_id == evc.uni_a.interface_id && request.uni_a.tag.value == evc.uni_a.tag.value);
+                var request_port_a_InUse2 = (request.uni_a.interface_id == evc.uni_z.interface_id && request.uni_a.tag.value == evc.uni_z.tag.value);
+
+                //is request endpoint Z port in used
+                var request_port_z_InUse1 = (request.uni_z.interface_id == evc.uni_a.interface_id && request.uni_z.tag.value == evc.uni_a.tag.value);
+                var request_port_z_InUse2 = (request.uni_z.interface_id == evc.uni_z.interface_id && request.uni_z.tag.value == evc.uni_z.tag.value);
```

Comment:
Does this logic need to be here, or can it be in the MEF E-line backend?

Reply:
@acard112, this check is being implemented on PR #220. Please have a look there and see if it solves this problem for you. If yes, you can just remove this test from your client code.